proof/proof.go

```diff
@@ -124,6 +130,27 @@ const (
 	// PrefixMagicBytesLength is the length of the magic bytes that are
 	// prefixed to individual proofs or proof files.
 	PrefixMagicBytesLength = 4
+
+	// MaxNumTaprootProofs is the maximum number of Taproot proofs there can
+	// be in a proof. This limit represents the maximum block size in vBytes
+	// divided by the size of a single P2TR output and is therefore only a
+	// theoretical limit that can never be reached in practice.
+	MaxNumTaprootProofs uint64 = blockchain.MaxBlockBaseSize /
+		input.P2TRSize
+
+	// MaxTaprootProofSizeBytes is the maximum size of a single Taproot
+	// proof. A Taproot proof can contain a commitment proof which at
+	// maximum can contain two MS-SMT proofs that max out at around 10k
+	// bytes each (in the worst case).
+	MaxTaprootProofSizeBytes = tlv.MaxRecordSize
+
+	// BlockHeaderSizeBytes is the size of a Bitcoin block header in bytes.
+	BlockHeaderSizeBytes = 80
+
+	// MerkleProofMaxNodes is the maximum number of nodes a merkle proof can
+	// contain. This is log2(max_num_txs_in_block) + 1, where max number of
+	// transactions in a block is assumed to be 17k (theoretical value).
```

Comment:
Would be good to have more explanation / documentation here on where the 17k comes from; I saw slightly different numbers from a cursory search (but the end result was still 14 or 15).